src/napari_clusters_plotter/_new_plotter_widget.py

```diff
@@ -170,25 +177,37 @@ def _replot(self):
 
         # if no x or y axis is selected, return
         if self.x_axis == "" or self.y_axis == "":
+            self._clear_plot_data()
             return
 
         # retrieve the data from the selected layers
         features = self._get_features()
         x_data = features[self.x_axis].values
         y_data = features[self.y_axis].values
 
+        active_artist = self.plotting_widget.active_artist
+
         # check hue axis for categorical data
         if self.hue_axis in self.categorical_columns:
-            self.plotting_widget.active_artist.overlay_colormap = (
-                cat10_mod_cmap
-            )
+            if (
+                active_artist
+                == self.plotting_widget.artists[ArtistType.SCATTER]
+            ):
+                self.plotting_widget.active_artist.overlay_colormap = (
+                    cat10_mod_cmap
+                )
+            elif (
+                active_artist
+                == self.plotting_widget.artists[ArtistType.HISTOGRAM2D]
+            ):
+                self.plotting_widget.active_artist.overlay_colormap = (
+                    cat10_mod_cmap_first_transparent
+                )
```

Comment:
Instead of the long `if/else` statement, maye it would be easier to create a dictionary like:

```python
colormap_dict = {
  ArtistType.SCATTER: cat10_mod_cmap,
  ArtistType.HISTOGRAM2D: cat10_mod_cmap_first_transparent
}
```
Here, the colormap could be set simply by

```
self.plotting_widget.active_artist.overlay_colormap = colormap_dict[type(active_artist])
```

and keep the code a bit more concise?

Reply:
Sure!